src/libraries/System.Private.CoreLib/src/System/Environment.Unix.cs

```diff
@@ -15,7 +15,9 @@ namespace System
 {
     public static partial class Environment
     {
+#if !TARGET_IOS
```

Comment:
We've tried to avoid lots of ifdefing in these files.  Can we refactor this to use partials?  That looks like it'd be trivial for at least some things, like GetFolderPathCoreWithoutValidation.

Reply:
@stephentoub I've just moved `GetFolderPathCore` with private dependencies to a new file -`Environment.Unix.GetFolderPathCore.cs `.  Environment.Unix.cs already contained a few if-defs, I'll try to get rid of them in a separate commit

Reply:
Thanks!

Reply:
(oops, I can't use `RuntimeInformation.IsOSPlatform` in corlib)